contrib/.vscode/launch.json

```diff
@@ -0,0 +1,94 @@
+{
+  "version": "0.2.0",
+  "configurations": [
+    {
+      "name": "Debug C++ Tests",
+      "type": "cppdbg",
+      "request": "launch",
+      "cwd": "${workspaceFolder}",
+      "program": "${command:cmake.buildDirectory}/tests/Tests_run", 
+      "args": [],
+      "environment": [
+        {
+          "name": "PATH",
+          "value": "${command:cmake.buildDirectory}/tests:${env:PATH}"
+        }
+      ],
+      "linux": {
+        "MIMode": "gdb",
+        "miDebuggerPath": "/usr/bin/gdb"
+      },
+      "osx": {
+        "MIMode": "lldb"
+      },
+      "stopAtEntry": false,
+      "externalConsole": false,
+      "preLaunchTask": "CMake: build",
+      "sourceFileMap": {
+        "${workspaceFolder}": "${workspaceFolder}"
+      }
+    },
+    {
+      "name": "C/C++ debugger",
+      "type": "cppdbg",
+      "request": "launch",
+      "cwd": "${workspaceFolder}",
+      "program": "${command:cmake.buildDirectory}/bin/FreeCAD", 
+      "args": ["${workspaceFolder}/.vscode/scripts/VSCodeAutostartDebug.FCMacro"],
+      "environment": [
+        {
+          "name": "PATH",
+          "value": "${command:cmake.buildDirectory}/bin:${env:PATH}"
+        },
+        {
+          "name": "PYDEVD_DISABLE_FILE_VALIDATION",
+          "value": "1"
+        }
+      ],
+      "linux": {
+        "MIMode": "gdb",
+        "miDebuggerPath": "/usr/bin/gdb"
+      },
+      "osx": {
+        "MIMode": "lldb"
+      },
+      "stopAtEntry": false,
+      "externalConsole": false,
+      "presentation": {
+        "hidden": true,
+      }
+    },
+    {
+      "name": "Python debugger",
```

Comment:
I will again raise my disagreement on this approach.   FreeCAD has built-in support for launching the VS Code debugger protocol, so having a launch task that uses another approach is superfluous.  Instead, there should be a launch task to attach to the FreeCAD running the VS Code debugger protocol.

Furthermore, having the task launch FreeCAD means that FreeCAD cannot also, simultaneously be executing via the C++ debugger.